index.js

```diff
@@ -265,7 +265,13 @@ class ServerlessAmplifyPlugin {
             }
         }
 
-        const identityPool = resources.find(r => r.ResourceType === 'AWS::Cognito::IdentityPool');
+        const identityPools = resources.filter(r => r.ResourceType === 'AWS::Cognito::IdentityPool');
+        const identityPoolForUserPool = fileDetails.hasOwnProperty('appClient')
+            ? identityPools.find(r => r.metadata.CognitoIdentityProviders &&
+                r.metadata.CognitoIdentityProviders.some(({ClientId}) => ClientId === config.CognitoUserPool.Default.AppClientId)
+            )
+            : undefined;
+        const identityPool = identityPoolForUserPool || identityPools[0];
```

Comment:
What happens here if you don't have an identity pool?  I'm thinking that because identityPoolForUserPool is undefined, it will flow into identityPools[0] and that will throw an index out of bounds (i.e. an error, rather than undefined).

I think making this:
```
const identityPool = identityPoolForUserPool || identityPools.length > 0 ? identityPools[0] : undefined;
```

would work instead?

Reply:
There is no such thing as an index out of bounds error in JS. If the element does not exist, is simply returns `undefined` (which is what I am going for here). Additionally, having a length > 0 does not guarantees the existence of the indices (in sparse arrays, which this one is not).

I verified this behavior on Chrome, Safari and NodeJS v10.11.0, with and without 'use strict'.